### PR TITLE
feat: add try/from trait for conversion between chrono and Duration

### PR DESCRIPTION
**Pull request by mpeddada1**
For #423.

Add TryFrom trait for conversion between [chrono::time](https://crates.io/crates/chrono) and `Duration`.

### CODE REVIEW

File: src/wkt/src/duration.rs [hunk 1]
```diff
@@ -272,6 +272,28 @@ impl std::convert::From<Duration> for time::Duration {
...
+/// Converts from [Duration] to [chrono::Duration].
+#[cfg(feature = "chrono")]
+impl std::convert::TryFrom<Duration> for chrono::Duration {
+    type Error = DurationError;
+    fn try_from(value: Duration) -> Result<Self, Self::Error> {
+        if value.nanos < 0 {
+            return Err(Error::OutOfRange());
+        }
+        Ok(Self::new(value.seconds(), value.nanos() as u32).unwrap())
+    }
```
Comment (coryan) [inline src/wkt/src/duration.rs:294]: I think you can safely handle the `value.nanos < 0` case using:

```suggestion
        Ok(Self::seconds(value.seconds()) + Self::nanoseconds(value.nanos()))
    }
```

Which suggests you do not need a `TryFrom` at all, and a simple `From` will do.

File: src/wkt/Cargo.toml [hunk 2]
```diff
@@ -24,17 +24,19 @@ keywords.workspace   = true
...
 [features]
+chrono = []
 time = []
 
 [dependencies]
 serde      = { version = "1.0.214", features = ["serde_derive"] }
 serde_with = { version = "3.11.0", default-features = false, features = ["base64"] }
 serde_json = "1.0.134"
 time       = { version = "0.3.36", features = ["formatting", "parsing"] }
+chrono     = { version = "0.4.39" }
```
Comment (coryan) [inline src/wkt/Cargo.toml:35]: I think you can say

```suggestion
chrono     = { version = "0.4.39", optional = true }
```

and in the features section:

```
chrono = ["dep:chrono"]
```

**Review by coryan — CHANGES_REQUESTED** [review]

File: src/wkt/src/duration.rs (hunk 1 above)
Comment (mpeddada1) [reply src/wkt/src/duration.rs]: That's a good idea!

File: src/wkt/Cargo.toml (hunk 2 above)
Comment (mpeddada1) [reply src/wkt/Cargo.toml]: Sounds good, made chrono an optional dependency.

**Review by coryan — APPROVED** [review]

**Pull request merged**